fdbcli/TenantCommands.actor.cpp

```diff
@@ -167,24 +167,37 @@ void applyConfigurationToSpecialKeys(Reference<ITransaction> tr,
 
 // tenant create command
 ACTOR Future<bool> tenantCreateCommand(Reference<IDatabase> db, std::vector<StringRef> tokens) {
-	if (tokens.size() < 3 || tokens.size() > 5) {
-		fmt::print("Usage: tenant create <NAME> [tenant_group=<TENANT_GROUP>] [assigned_cluster=<CLUSTER_NAME>]\n\n");
+	if (tokens.size() < 3 || tokens.size() > 6) {
+		fmt::print("Usage: tenant create <NAME> [tenant_group=<TENANT_GROUP>] [assigned_cluster=<CLUSTER_NAME>] "
+		           "[ignore_capacity_limit]\n\n");
 		fmt::print("Creates a new tenant in the cluster with the specified name.\n");
 		fmt::print("An optional group can be specified that will require this tenant\n");
 		fmt::print("to be placed on the same cluster as other tenants in the same group.\n");
 		fmt::print("An optional cluster name can be specified that this tenant will be placed in.\n");
+		fmt::print("Optionally, `ignore_capacity_limit' can be specified together with `assigned_cluster' to create a "
+		           "tenant ignoring tenant group limit on the specified cluster.\n");
 		return false;
 	}
 
 	state Key tenantNameKey = tenantMapSpecialKeyRange.begin.withSuffix(tokens[2]);
 	state Reference<ITransaction> tr = db->createTransaction();
 	state bool doneExistenceCheck = false;
 
+	state bool ignoreCapacityLimit = tokens.back() == "ignore_capacity_limit";
+	int configurationEndIndex = tokens.size() - (ignoreCapacityLimit ? 1 : 0);
+
 	state Optional<std::map<Standalone<StringRef>, Optional<Value>>> configuration =
-	    parseTenantConfiguration(tokens, 3, tokens.size(), false);
+	    parseTenantConfiguration(tokens, 3, configurationEndIndex, false);
 
 	if (!configuration.present()) {
 		return false;
+	} else if (ignoreCapacityLimit) {
+		if (!configuration.get().contains("assigned_cluster"_sr) && !configuration.get().contains("tenant_group"_sr)) {
```

Comment:
We should disallow ignoring the capacity limit if just `assigned_cluster` isn't specified. Whether or not a tenant group is not sufficient to know whether a tenant group will be created. We can also combine these two if statements into 1.

Reply:
> Whether or not a tenant group is not sufficient to know whether a tenant group will be created

Right, thus I think we need to be cautious not to return error too early. We should return error if `assigned_cluster` is not specified AND the tenant group does not exist. At this point, we don't know whether the tenant group (if specified) exists or not, but we do know if tenant group is not specified and `assigned_cluster` is not specified, it must be disallowed. Later, we perform check for the existence of tenant group within `metacluster` lib.

Reply:
On a second thought, I agree that we should disallow ignoring capacity limit as long as `assigned_cluster` is not specified. In the case of creating a tenant in an existing group without specifying assigned cluster, capacity limit is already honored, and `ignore_capacity_limit` is inaccurate description of the behavior.

Reply:
Done